py/server/deephaven/updateby.py

```diff
@@ -164,6 +171,211 @@ def ema_time_decay(ts_col: str, time_scale: Union[int, str], cols: Union[str, Li
+    the decay unit.
+
+    The formula used is
+        a = e^(-1 / time_scale_ticks)
+        ems_next = a * ems_last + value
+
+    Args:
+        time_scale_ticks (int): the decay rate in ticks
+        cols (Union[str, List[str]]): the column(s) to be operated on, can include expressions to rename the output,
+            i.e. "new_col = col"; when empty, update_by perform the ems operation on all columns.
+        op_control (OperationControl): defines how special cases should behave, when None, the default OperationControl
+            settings as specified in :meth:`~OperationControl.__init__` will be used
+
+    Returns:
+        an UpdateByOperation
+
+    Raises:
+        DHError
+    """
+    try:
+        cols = to_sequence(cols)
+        if op_control is None:
+            return UpdateByOperation(j_updateby_op=_JUpdateByOperation.Ems(time_scale_ticks, *cols))
+        else:
+            return UpdateByOperation(
+                j_updateby_op=_JUpdateByOperation.Ems(op_control.j_op_control, time_scale_ticks, *cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a tick-decay EMS UpdateByOperation.") from e
+
+def ems_time_decay(ts_col: str, time_scale: Union[int, str], cols: Union[str, List[str]],
+                   op_control: OperationControl = None) -> UpdateByOperation:
+    """Creates an EMS (exponential moving sum) UpdateByOperation for the supplied column names, using time as the
+    decay unit.
+
+    The formula used is
+        a = e^(-dt / time_scale_nanos)
+        eems_next = a * ems_last + value
+
+     Args:
+        ts_col (str): the column in the source table to use for timestamps
+        time_scale (Union[int, str]): the decay rate, can be expressed as an integer in nanoseconds or a time
+            interval string, e.g. "00:00:00.001"
+        cols (Union[str, List[str]]): the column(s) to be operated on, can include expressions to rename the output,
+            i.e. "new_col = col"; when empty, update_by perform the ems operation on all columns.
+        op_control (OperationControl): defines how special cases should behave,  when None, the default OperationControl
+            settings as specified in :meth:`~OperationControl.__init__` will be used
+
+    Returns:
+        an UpdateByOperation
+
+    Raises:
+        DHError
+     """
+    try:
+        time_scale = _JDateTimeUtils.expressionToNanos(time_scale) if isinstance(time_scale, str) else time_scale
+        cols = to_sequence(cols)
+        if op_control is None:
+            return UpdateByOperation(j_updateby_op=_JUpdateByOperation.Ems(ts_col, time_scale, *cols))
+        else:
+            return UpdateByOperation(
+                j_updateby_op=_JUpdateByOperation.Ems(op_control.j_op_control, ts_col, time_scale, *cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a time-decay EMS UpdateByOperation.") from e
+
+
+def emmin_tick_decay(time_scale_ticks: int, cols: Union[str, List[str]],
```

Comment:
These new methods suffer from the same problem as the ticket I have reported twice.  The timescale should *not* be int.  It should be float.

I wasn't able to find the other ticket to link to.  Maybe it was fixed?